domain/application/state/application.go

```diff
@@ -245,41 +254,38 @@ func (st *ApplicationState) DeleteApplication(ctx context.Context, name string)
 		return errors.Trace(err)
 	}
 
-	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		appUUID, err := st.lookupApplication(ctx, tx, name, true)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		appID.ID = appUUID.String()
+	appUUID, err := st.lookupApplication(ctx, tx, name, true)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	appID.ID = appUUID.String()
 
-		// Check that there are no units.
-		result := sqlair.M{}
-		err = tx.Query(ctx, queryUnitsStmt, appID).Get(&result)
-		if err != nil {
-			if !errors.Is(err, sqlair.ErrNoRows) {
-				return errors.Annotatef(err, "querying units for application %q", name)
-			}
-		}
-		numUnits, _ := result["count"].(int64)
-		if numUnits > 0 {
-			return fmt.Errorf("cannot delete application %q as it still has %d unit(s)%w", name, numUnits, errors.Hide(applicationerrors.ApplicationHasUnits))
+	// Check that there are no units.
+	result := sqlair.M{}
+	err = tx.Query(ctx, queryUnitsStmt, appID).Get(&result)
+	if err != nil {
+		if !errors.Is(err, sqlair.ErrNoRows) {
```

Comment:
Don't think you can get a `sqlair.ErrNoRows` error, but it doesn't hurt to have it here.

Reply:
removed it